rpmlb/builder/koji.py

```diff
@@ -0,0 +1,173 @@
+"""Builder interface for the Koji build service."""
+
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Sequence
+
+from ..utils import run_cmd, run_cmd_with_capture
+from ..work import Work
+from .base import BaseBuilder
+
+
+class KojiBuilder(BaseBuilder):
+    """Builder interface for the Koji build service."""
+
+    DEFAULT_TARGET_FORMAT = 'sclo{epel}-{collection}-rh-el{epel}'
+
+    def __init__(self):
+        """Initialize the slots"""
+
+        self.collection = None  #: The collection to build
+        self.epel = None  #: Version of EPEL to build for
+        self.profile = None  #: The koji configuration profile to use
+        #: The format string for construction of the actual template name
+        self.target_format = type(self).DEFAULT_TARGET_FORMAT
+
+    @property
+    def base_command(self) -> Sequence[str]:
```

Comment:
Bit of a context from my side:

After in-person discussion about it, the fear that @junaruga had about this extending of subclass API was that it will confuse users. At the end we agreed that it is not an issue for us and decided to leave this properties "public" and leave the usage responsibility to user of the class.

@junaruga, feel free to add anything you deem necessary. From my point of view, this issue was addressed and resolved.